taskcoordinator/app/src/test/java/com/microservices/taskcoordinator/controllers/CoordinatorControllerIntegrationTests.java

```diff
@@ -77,12 +85,113 @@ public void testCoordinateOrder_allValid_orderCoordinatedAndLaundryStateChanged(
 
-        Assert.assertEquals(coordinationDto.getOrderId(), orderSubmissionDto.getOrderId());
-        Assert.assertEquals(orderSubmissionDto.getOrderId(), orderCreated.getId());
-        Assert.assertEquals(OrderStatus.RESERVED, orderCreated.getStatus());
+        assertEquals(coordinationDto.getOrderId(), orderSubmissionDto.getOrderId());
+        assertEquals(orderSubmissionDto.getOrderId(), orderCreated.getId());
+        assertEquals(OrderStatus.RESERVED, orderCreated.getStatus());
 
         long reserveDiff = laundryStateForCreatedOrder.getReservedTime() - laundryWillBeChosen.getReservedTime();
-        Assert.assertEquals((long) orderCreated.getDuration(), reserveDiff);
+        assertEquals((long) orderCreated.getDuration(), reserveDiff);
     }
 
+    @Test
+    @Sql(scripts = "/test-data/one_laundry.sql")
+    @Transactional
+    public void testSubmitOrder_fromReservedStatus() throws Exception {
+        int orderId = 1;
+        int laundryId = 1;
+
+        InboundLaundryStateDto inboundLaundryStateDto = new InboundLaundryStateDto(laundryId, 200L, 2);
+        OrderSubmittedDto orderSubmittedDto = new OrderSubmittedDto(orderId, inboundLaundryStateDto);
+
+        String orderSubmittedJson = objectMapper.writeValueAsString(orderSubmittedDto);
+        mockMvc.perform(put("/orders/666/status/submitted")
+                .contentType(APPLICATION_JSON.toString())
+                .content(orderSubmittedJson))
+                .andExpect(status().isOk());
+
+        LaundryStateDto updatedLaundryState = laundryStateService.getLaundryStateById(laundryId);
+
+        assertEquals(200, updatedLaundryState.getQueueWaitingTime().longValue());
+        assertEquals(100, updatedLaundryState.getReservedTime().longValue());
+        assertEquals(2, updatedLaundryState.getVersion().intValue());
+        assertEquals(OrderStatus.SUBMITTED, orderService.getOrderById(orderId).getStatus());
+    }
+
+    @Test
+    @Sql(scripts = "/test-data/basic.sql")
+    @Transactional
+    public void testProcessOrder_fromSubmittedStatus() throws Exception {
+        int orderId = 3;
+        int laundryId = 1;
+
+        InboundLaundryStateDto inboundLaundryStateDto = new InboundLaundryStateDto(laundryId, 200L, 2);
+        OrderProcessedDto orderSubmittedDto = new OrderProcessedDto(orderId, inboundLaundryStateDto, 102L);
+
+        String orderSubmittedJson = objectMapper.writeValueAsString(orderSubmittedDto);
+        mockMvc.perform(put("/orders/666/status/processed")
+                .contentType(APPLICATION_JSON.toString())
+                .content(orderSubmittedJson))
+                .andExpect(status().isOk());
+
+        LaundryStateDto updatedLaundryState = laundryStateService.getLaundryStateById(laundryId);
+
+        assertEquals(200, updatedLaundryState.getQueueWaitingTime().longValue());
+        assertEquals(100, updatedLaundryState.getReservedTime().longValue());
```

Comment:
why 100? why 200? count this values inside test (obviously values here can't be constants)